charts/policy-reporter/charts/kyvernoPlugin/values.yaml

```diff
@@ -195,3 +195,10 @@ global:
   namespace: ""
   # additional labels added on each resource
   labels: {}
+  # basicAuth for APIs and metrics
+  basicAuth:
+    # HTTP BasicAuth username
+    username: ""
+    # HTTP BasicAuth password
+    password: ""
```

Comment:
Hi @fjogeleit 

Thank you a lot
I just have one proposal,
Could you add functionality to read creds from existing secret?
```
  basicAuth:
    # HTTP BasicAuth username
    username: ""
    # HTTP BasicAuth password
    password: ""
```
For example
```
  basicAuth:
    # HTTP BasicAuth username
    username: ""
    # HTTP BasicAuth password
    password: ""
    existingSecret: "MySecretCreatedByHands'
```
In this case, the values.yaml will not keep sensitive data, another way, configure ENV variable read those parameters from ENV, in this case we can mount secret into ENV then application read creds from ENV

Reply:
thanks, I will check it

Reply:
You want to create the secret on your own or could it be created by the helm chart?

Reply:
my own

Reply:
I added `secretRef` to the basicAuth config. Its a name of an existing secret with a `username` and `password` field.

PolicyReporter Core App will load it at runtime to configure the BasicAuth. UI will configure it as ENV vars to do authenticated calls. The Plugin will also load it as env var to add authentication to its APIs.

If monitoring is enabled, ServiceMonitors will use the existing secret as well to configure basicAuth.

But you need to configure them explicit in the UI for external APIs because there is no way to access secrets in external services and I don't see a good way to configure them as ENV vars.

Reply:
https://github.com/kyverno/policy-reporter-ui/pull/295 Adds support for Authenticated calls, credentials can be provided as secret or directly in the config

Reply:
https://github.com/kyverno/policy-reporter-kyverno-plugin/pull/27 BasicAuth support for the KyvernoPlugin

I will now test the configurations and helm chart changes. The next release will end of the week or during the next week.